tests/manage/pv_services/pvc_clone/test_clone_when_pvc_full.py

```diff
@@ -0,0 +1,301 @@
+
+        # Attach the cloned PVCs to pods
+        log.info("Attach the cloned PVCs to pods")
+        clone_pod_objs = []
+        for clone_pvc_obj in cloned_pvcs:
+            interface = constants.CEPHFILESYSTEM if (
+                constants.CEPHFS_INTERFACE in clone_pvc_obj.backed_sc
+            ) else constants.CEPHBLOCKPOOL
+            clone_pod_obj = pod_factory(
+                interface=interface, pvc=clone_pvc_obj,
+                status=''
+            )
+            log.info(
+                f"Attached the PVC {clone_pvc_obj.name} to pod "
+                f"{clone_pvc_obj.name}"
+            )
+            clone_pod_objs.append(clone_pod_obj)
+
+        # Verify the new pods are running
+        log.info("Verify the new pods are running")
+        for pod_obj in clone_pod_objs:
+            wait_for_resource_state(pod_obj, constants.STATUS_RUNNING)
+        log.info("Verified: New pods are running")
+
+        # Verify that the md5sum matches
+        for pod_obj in clone_pod_objs:
+            log.info(
+                f"Verifying md5sum of {file_name} "
+                f"on pod {pod_obj.name}"
+            )
+            pod.verify_data_integrity(
+                pod_obj,
+                file_name,
+                pod_obj.pvc.parent.md5sum
+            )
+            log.info(
+                f"Verified: md5sum of {file_name} on pod {pod_obj} "
+                f"matches with the original md5sum"
+            )
+
+        # Wait till utilization alerts starts
+        for response in TimeoutSampler(180, 5, prometheus_api.get, 'alerts'):
+            alerts = response.json()['data']['alerts']
+            for pvc_obj in cloned_pvcs:
+                alerts_pvc = [
+                    alert for alert in alerts if alert.get('labels', {}).get(
+                        'persistentvolumeclaim'
+                    ) == pvc_obj.name
+                ]
+                # At least 2 alerts should be present
+                if len(alerts_pvc) < 2:
+                    break
+
+                # Verify 'PersistentVolumeUsageNearFull' alert is firing
+                if not getattr(pvc_obj, 'near_full_alert', False):
+                    try:
+                        log.info(
+                            f"Checking 'PersistentVolumeUsageNearFull' alert "
+                            f"for PVC {pvc_obj.name}"
+                        )
+                        near_full_msg = (
+                            f"PVC {pvc_obj.name} is nearing full. Data "
+                            f"deletion or PVC expansion is required."
+                        )
+                        check_alert_list(
+                            label='PersistentVolumeUsageNearFull',
+                            msg=near_full_msg, alerts=alerts_pvc,
+                            states=['firing'], severity='warning'
+                        )
+                        pvc_obj.near_full_alert = True
+                    except AssertionError:
```

Comment:
shouldn't we raise the exception in this case?

Reply:
No, retry will work until timeout.